packages/hash/frontend/src/blocks/page/Comments/CreateBlockCommentButton.tsx

```diff
@@ -1,42 +1,37 @@
-import { FunctionComponent, useCallback, useState } from "react";
+import { FunctionComponent, useCallback, useState, useRef } from "react";
 import { IconButton, FontAwesomeIcon } from "@hashintel/hash-design-system";
 import { faComment } from "@fortawesome/free-regular-svg-icons";
 import Box from "@mui/material/Box";
-import { TextToken } from "@hashintel/hash-shared/graphql/types";
 import Popper from "@mui/material/Popper";
-import { useCreateComment } from "../../../components/hooks/useCreateComment";
-import { useRouteAccountInfo } from "../../../shared/routing";
-import { CommentTextField } from "./CommentTextField";
 import styles from "../style.module.css";
+import { CreateBlockComment } from "./CreateBlockComment";
+import { CommentTextFieldRef } from "./CommentTextField";
 
-type CommentButtonProps = {
+type CreateBlockCommentButtonProps = {
   blockId: string | null;
   rootNode: HTMLElement;
 };
 
-export const CommentButton: FunctionComponent<CommentButtonProps> = ({
-  blockId,
-  rootNode,
-}) => {
-  const { accountId } = useRouteAccountInfo();
+export const CreateBlockCommentButton: FunctionComponent<
+  CreateBlockCommentButtonProps
+> = ({ blockId, rootNode }) => {
+  const inputRef = useRef<CommentTextFieldRef>(null);
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
-  const [createComment] = useCreateComment(accountId);
 
-  const submitComment = useCallback(
-    async (content: TextToken[]) => {
-      if (blockId) {
-        await createComment(blockId, content);
-      }
-    },
-    [createComment, blockId],
-  );
-
-  const closeInput = useCallback(() => setAnchorEl(null), []);
+  const closeInput = useCallback(() => {
+    inputRef.current?.resetDocument();
```

Comment:
Removed `keepMounted` on the popper that renders `CreateBlockComment` so that it is destroyed when the popper is closed. b8f14a718fac78feb592f3fad6c53c58a9858e05